src/gui.cpp

```diff
@@ -128,24 +128,62 @@ namespace big
 		colors[ImGuiCol_PlotHistogramHovered] = ImVec4(0.25f, 1.00f, 0.00f, 1.00f);
 		colors[ImGuiCol_TextSelectedBg]       = ImVec4(0.25f, 1.00f, 0.00f, 0.43f);
 
-		memcpy(&m_default_config, &ImGui::GetStyle(), sizeof(ImGuiStyle));
+		save_default_style();
 	}
 
 	void gui::dx_on_tick()
 	{
 		if (m_is_open)
 		{
-			ImGui::PushStyleColor(ImGuiCol_WindowBg, ImGui::ColorConvertU32ToFloat4(g.window.color));
-			view::root();
-			ImGui::PopStyleColor();
+			push_theme_colors();
+			view::root(); // frame bg
+			pop_theme_colors();
 		}
 	}
 
+	void gui::save_default_style()
+	{
+		memcpy(&m_default_config, &ImGui::GetStyle(), sizeof(ImGuiStyle));
+	}
+
 	void gui::restore_default_style()
 	{
 		memcpy(&ImGui::GetStyle(), &m_default_config, sizeof(ImGuiStyle));
 	}
 
+	void gui::push_theme_colors()
+	{
+		auto button_color = ImGui::ColorConvertU32ToFloat4(g.window.button_color);
+		auto button_active_color =
+		    ImVec4(button_color.x + 0.33f, button_color.y + 0.33f, button_color.z + 0.33f, button_color.w);
+		auto frame_color = ImGui::ColorConvertU32ToFloat4(g.window.frame_color);
+		auto frame_hovered_color =
+		    ImVec4(frame_color.x + 0.14f, frame_color.y + 0.14f, frame_color.z + 0.14f, button_color.w);
+		auto frame_active_color =
+		    ImVec4(frame_color.x + 0.30f, frame_color.y + 0.30f, frame_color.z + 0.30f, button_color.w);
+
+		ImGui::PushStyleColor(ImGuiCol_WindowBg, ImGui::ColorConvertU32ToFloat4(g.window.background_color));
+		ImGui::PushStyleColor(ImGuiCol_Text, ImGui::ColorConvertU32ToFloat4(g.window.text_color));
+		ImGui::PushStyleColor(ImGuiCol_Button, button_color);
+		ImGui::PushStyleColor(ImGuiCol_ButtonHovered, button_color);
+		ImGui::PushStyleColor(ImGuiCol_ButtonActive, button_active_color);
+		ImGui::PushStyleColor(ImGuiCol_FrameBg, frame_color);
+		ImGui::PushStyleColor(ImGuiCol_FrameBgHovered, frame_hovered_color);
+		ImGui::PushStyleColor(ImGuiCol_FrameBgActive, frame_active_color);
+	}
+
+	void gui::pop_theme_colors()
+	{
+		ImGui::PopStyleColor();
```

Comment:
this whole part scream for a wrapper function that also increment a variable which is then used for forloop